packages/merge-conflicts/src/browser/merge-conflicts-parser.ts

```diff
@@ -0,0 +1,203 @@
+        currentMarker.to(separator, startsWithEq);
+        currentMarker.to(currentContent, any);
+        currentContent.to(currentMarker, startsWithLt);
+        currentContent.to(separator, startsWithEq);
+        currentContent.to(baseMarker, startsWithPp);
+        currentContent.to(start, startsWithGt);
+        currentContent.to(currentContent, any);
+        baseMarker.to(currentMarker, startsWithLt);
+        baseMarker.to(baseMarker, startsWithPp);
+        baseMarker.to(separator, startsWithEq);
+        baseMarker.to(baseContent, any);
+        baseContent.to(currentMarker, startsWithLt);
+        baseContent.to(separator, startsWithEq);
+        baseContent.to(baseMarker, startsWithPp);
+        baseContent.to(baseContent, any);
+        separator.to(currentMarker, startsWithLt);
+        separator.to(start, startsWithEq);
+        separator.to(incomingContent, any);
+        incomingContent.to(start, startsWithEq);
+        incomingContent.to(currentMarker, startsWithLt);
+        incomingContent.to(incomingMarker, startsWithGt);
+        incomingContent.to(incomingContent, any);
+        incomingMarker.to(push);
+        push.to(start);
+
+        // actions
+        currentMarker.onEnter = (input, ctx) => {
+            ctx.new = new Context().new;
+            ctx.new.current.marker = this.lineToRange(input);
+        };
+        currentContent.onEnter = (input, ctx) => {
+            const current = ctx.new.current;
+            current.content = this.addLineToRange(input, current.content);
+        };
+        baseMarker.onEnter = (input, ctx) => {
+            ctx.new.bases.push({ marker: this.lineToRange(input) });
+        };
+        baseContent.onEnter = (input, ctx) => {
+            const base = ctx.new.bases.slice(-1)[0];
+            base.content = this.addLineToRange(input, base.content);
+        };
+        incomingContent.onEnter = (input, ctx) => {
+            const incoming = ctx.new.incoming;
+            incoming.content = this.addLineToRange(input, incoming.content);
+        };
+        incomingMarker.onEnter = (input, ctx) => {
+            const markerRange = this.lineToRange(input);
+            ctx.new.incoming.marker = markerRange;
+            ctx.new.total = {
+                start: ctx.new.current.marker!.start,
+                end: markerRange.end
+            };
+        };
+        push.onEnter = (input, ctx) => {
+            ctx.all.push(ctx.new);
+        };
+    }
+
+    private lineToRange(input: Input): Range {
+        return {
+            start: { line: input.line, character: 0 },
+            end: { line: input.line, character: input.text.length },
+        };
+    }
+
+    private addLineToRange(input: Input, range: Range | undefined): Range {
+        if (!range) {
+            return this.lineToRange(input);
+        }
+        range.end = { line: input.line, character: input.text.length };
+        return range;
+    }
+}
+
+class Context {
```

Comment:
It's up to you here. I would put internal classes in `MergeConflictsParser` namespace and reexport just in case if someone wants to customize `MergeConflictsParser` or temporary workaround a bug.

Reply:
moving to NS.